examples/failing/InstanceChainBothUnknownAndMatch.purs

```diff
@@ -1,14 +1,18 @@
 -- @shouldFailWith NoInstanceFound
 module InstanceChains.BothUnknownAndMatch where
 
-class Same l r o | l r -> o
-instance sameY :: Same t t @"Y" else instance sameN :: Same l r @"N"
-same :: forall l r o. Same l r o => l -> r -> @o
-same _ _ = @o
+import Type.Proxy (Proxy)
+import Type.Row (RProxy(..))
+import Data.Symbol (SProxy(..))
+
+class Same l r (o :: Symbol) | l r -> o
+instance sameY :: Same t t "Y" else instance sameN :: Same l r "N"
```

Comment:
@LiamGoodacre I've changed this class slightly, so that the `o` parameter is a `Symbol` rather than a proxy, just because that seemed to make it slightly simpler. I did the same in the other failing instance chains test as well. Do you think this makes sense?

Reply:
Yes, that makes sense.